contracts/Stake/RewardDelegators.sol

```diff
@@ -156,122 +147,108 @@ contract RewardDelegators is Initializable, Ownable {
                                                                 );
             }
         }
-        clusters[_cluster].lastRewardDistNonce = cluster.lastRewardDistNonce.add(1);
         transferRewards(clusterRegistry.getRewardAddress(_cluster), commissionReward);
         emit ClusterRewardDistributed(_cluster);
     }
 
     function delegate(
-        address _delegator, 
-        address _cluster, 
-        bytes32[] memory _tokens, 
+        address _delegator,
+        address _cluster,
+        bytes32[] memory _tokens,
         uint256[] memory _amounts
     ) public onlyStake {
         _updateRewards(_cluster);
-        Cluster memory clusterData = clusters[_cluster];
-        require(
-            clusterRegistry.isClusterValid(_cluster),
-            "ClusterRegistry:delegate - Cluster should be registered to delegate"
-        );
-        uint256 currentNonce = clusterData.lastRewardDistNonce;
-        uint256 totalRewards;
-        uint256 totalRewardDebt;
-        for(uint256 i=0; i < _tokens.length; i++) {
-            uint256 tokenAccRewardPerShare = clusters[_cluster].accRewardPerShare[_tokens[i]];
-            uint256 delegatorTokens = clusters[_cluster].delegators[_delegator][_tokens[i]];
+
+        uint256 _pendingRewards;
+        for(uint256 i = 0; i < _tokens.length; i++) {
+            uint256 _amount = _amounts[i];
+            bytes32 _tokenId = _tokens[i];
+            if(_amount == 0) continue;
```

Comment:
May be move this to line 179. Let user withdraw rewards even when amount is 0 ?

Reply:
let them just call withdrawRewards na, why increase gas consumption for everybody here...

Reply:
should move it above 165 in fact...

Reply:
Yeah actually there will be more 0 calls now due to changes in stakeManager. Makes sense in that cases. Moving the if to 165.